http/codegen/client.go

```diff
@@ -466,3 +491,27 @@ const singleResponseT = `case {{ .StatusCode }}:
 			}
 	{{- end }}
 `
+
+// input: multipartData
+const multipartRequestEncoderTypeT = `{{ printf "%s is the type to encode multipart request for the %q service %q endpoint." .FuncName .ServiceName .MethodName | comment }}
+type {{ .FuncName }} func(*multipart.Writer, *{{ .PayloadRef }}) error
+`
+
+// input: multipartData
+const multipartRequestEncoderT = `{{ printf "%s returns an encoder to encode the multipart request for the %q service %q endpoint." .InitName .ServiceName .MethodName | comment }}
+func {{ .InitName }}(encoderFn {{ .FuncName }}) func(r *http.Request) goahttp.Encoder {
+	return func(r *http.Request) goahttp.Encoder {
+		body := &bytes.Buffer{}
+		mw := multipart.NewWriter(body)
+		return goahttp.EncodingFunc(func(v interface{}) error {
+			p := v.(*{{ .PayloadRef }})
```

Comment:
That hard-coded `*` doesn't seem right...

Reply:
Fixed